### PR TITLE
Add Rapidfire

### PR DESCRIPTION
**Pull request by BenCat07**
I guess i somehow made it work?

### CODE REVIEW

**Comment by BenCat07** [comment]
The weapon shooting cooldowns are still a bit wonky, i'd recommend disabling "Only when can shoot" in aimbot.

**Review by TotallyNotElite — CHANGES_REQUESTED** [review]

**Comment by TotallyNotElite** [comment]
Also, maybe doubleshot/doubletap is not the best term for this, since you can shoot more than twice

File: src/hacks/Warp.cpp [hunk 1]
```diff
@@ -238,12 +311,65 @@ static int charge_at_start = 0;
 
 // Used to determine when demoknight warp should be over
 static bool was_overridden = false;
+
+// How long since we are in one of the revved states?
+static int ticks_in_revved     = 0;
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Warp.cpp:318]: There is alot of jank related to how the game handles firing. Doubletap wall refuse to work if you spin up with m1, although it has the same state as m2 spinup. This is also a special case thing i will only use once, but yeah even if you spin up with m2 instead, you need to wait a bit after you spinned up to actually doubletap, it's extremely janky game mechanics.

File: src/hacks/Warp.cpp [hunk 2]
```diff
@@ -48,17 +62,60 @@ static bool charged             = false;
 static bool should_warp = true;
 static bool was_hurt    = false;
 
+bool shouldDoubleshot()
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Warp.cpp:68]: f

File: src/hacks/Warp.cpp [hunk 3]
```diff
@@ -48,17 +62,60 @@ static bool charged             = false;
...
 }
 
 // How many ticks of excess we have (for decimal speeds)
 float excess_ticks = 0.0f;
 int GetWarpAmount(bool finalTick)
 {
     int max_extra_ticks = *maxusrcmdprocessticks - 1;
+
+    // Doubletap ignores speed
+    if (in_doubletap)
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Warp.cpp:120]: yeah i'll take your commented suggestion and fix the naming

**Comment by BenCat07** [comment]
Renamed and fixed some really big issues

**Review by notroboot — APPROVED** [review]
Tested for a while and it seems to work well.

**Review by TotallyNotElite — APPROVED** [review]
You CANT make me review this code. I refuse. I demand a salary of 100k per year and benefits.

**Pull request merged**